### PR TITLE
[image_picker] Image picker select preferred camera device (rear, front)

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

Add a new feature to let user select preferred camera device.

## Related Issues

https://github.com/flutter/flutter/issues/24880

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/image_picker/lib/image_picker.dart [hunk 1]
```diff
@@ -23,6 +23,21 @@ enum ImageSource {
   gallery,
 }
 
+/// Which camera to use when picking images/videos while source is `ImageSource.camera`.
+///
+/// Not every device supports both of the positions.
+enum CameraDevice {
```
Comment (mehmetf) [inline packages/image_picker/lib/image_picker.dart]: I would name this `CameraFacing`.

or if you anticipate there will be more properties going into here, you can make it a class with const ctr.

File: packages/image_picker/lib/image_picker.dart [hunk 2]
```diff
@@ -79,16 +101,21 @@ class ImagePicker {
...
+  /// Defaults to [CameraDevice.rear].
+  ///
   /// In Android, the MainActivity can be destroyed for various fo reasons. If that happens, the result will be lost
   /// in this call. You can then call [retrieveLostData] when your app relaunches to retrieve the lost data.
-  static Future<File> pickVideo({
-    @required ImageSource source,
-  }) async {
+  static Future<File> pickVideo(
+      {@required ImageSource source,
+      CameraDevice preferredCameraDevice = CameraDevice.rear}) async {
```
Comment (mehmetf) [inline packages/image_picker/lib/image_picker.dart]: `preferredCameraFacing`

File: packages/image_picker/lib/image_picker.dart (hunk 1 above)
Comment (cyanglaz) [reply packages/image_picker/lib/image_picker.dart]: Yeah, `CameraFacing` is probably a more straight forward name for the current situation.
I named this `CameraDevice` following the iOS naming. I think `CameraDevice` is more future proof as there might be different cameras other than front and rear in future. Maybe like a secondary rear camera that can be chosen as default camera.
WDYT?

Comment (mehmetf) [reply packages/image_picker/lib/image_picker.dart]: Sure. It was a nit anyway.

**Comment by t1m9m** [comment]
@cyanglaz, I cannot thank you enough for this awesome integration; Thank you so much, really appreciated;

**Pull request merged**